dcc-portal-ui/app/scripts/pathwayviewer/js/pathways.js

```diff
@@ -110,252 +110,80 @@
-                _geneOverlapExistsHash[g] = true;
-              });
-              
-              
-              //console.log('Overlap genes returned: ' +
-              // _.keys(_geneOverlapExistsHash).length, '\nProtein Map Intersect Overlap Length = ' +
-              //  (_.intersection(_.values(_uniprotIds), _.keys(_geneOverlapExistsHash)).length) );
-              deferred.resolve();
-              
-            });
-          return deferred.promise;
-        })
-        .then(function () {
-          var deferred = $q.defer();
-          
-          GeneSetVerificationService.verify(_uniprotIds.join(','))
-            .then(function (data) {
-              
-              var geneCount = 0;
-              
-              _.forEach(_pathwayMutationHighlights, function (n) {
-                var geneKey = 'external_db_ids.uniprotkb_swissprot';
-                if (!data.validGenes[geneKey]) {
-                  return;
-                }
-                
-                var uniprotObj = data.validGenes[geneKey][n.uniprotId];
-                if (!uniprotObj) {
-                  return;
-                }
-                var ensemblId = uniprotObj[0].id;
-                n.advQuery = LocationService.mergeIntoFilters({
-                  gene: {
-                    id: {is: [ensemblId]}
-                  }
-                });
-                n.geneSymbol = uniprotObj[0].symbol;
-                n.geneId = ensemblId;
-                
-                if (angular.isDefined(_geneOverlapExistsHash[n.uniprotId])) {
-                  geneCount++;
-                  
-                  _.forEach(n.dbIds, function (dbID) {
-                    // Swap in Reactome keys but maintain the id we use this to determine overlaps in O(1)
-                    // later... The dbID is used as a reference to the reactome SVG nodes...
-                    _geneOverlapExistsHash[dbID] = {id: n.uniprotId, geneId: n.geneId, geneSymbol: n.geneSymbol};
-                  });
-                  delete _geneOverlapExistsHash[n.uniprotId];
-                }
-              });
-              console.log(geneCount + ' Overlapped genes validated! ');
-              deferred.resolve();
-            });
-          
-          return deferred.promise;
-        })
-        .then(function () {
-          $scope.geneSet = _geneSet;
+
+      PathwayDataService.getPathwayData(pathway.geneSetId, pathway.geneSetOverlapFilters)
+        .then(function (pathwayData) {
+          $scope.geneSet = pathwayData.geneSet;
           $scope.pathway = {
-              xml: _xml,
-              zooms: _zooms,
-              mutationHighlights: _pathwayMutationHighlights,
-              overlaps: _geneOverlapExistsHash
+            xml: pathwayData.xml,
+            zooms: pathwayData.zooms,
+            mutationHighlights: pathwayData.mutationHighlights,
+            drugHighlights: pathwayData.drugHighlights,
+            overlaps: pathwayData.overlaps
```

Comment:
Actually, why aren't we just assigning the value of pathwayData? Is it all the Restangular crap? Try 

```js
//..
.then(function (response) {
  $scope.pathway = response.plain();
})
```

Does that keep only what you need?